crates/core/src/share.rs

```diff
@@ -0,0 +1,106 @@
+use std::{collections::HashMap, convert::TryFrom, fs, path::Path};
+
+use serde::{Deserialize, Serialize};
+
+use crate::{addon::Addon, config::Flavor, error, repository::RepositoryKind};
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Data {
+    pub flavor: Flavor,
+    pub repo_kind: RepositoryKind,
+    pub id: String,
+    pub name: String,
+}
+
+impl TryFrom<(Flavor, Addon)> for Data {
+    type Error = ();
+
+    fn try_from((flavor, addon): (Flavor, Addon)) -> Result<Self, Self::Error> {
+        let name = addon.title().to_string();
+        let repository = addon.repository.ok_or(())?;
+        let repo_kind = repository.kind;
+        let id = repository.id;
+
+        Ok(Self {
+            flavor,
+            repo_kind,
+            id,
+            name,
+        })
+    }
+}
+
+pub fn export(
+    addons: HashMap<Flavor, Vec<Addon>>,
+    output_dir: impl AsRef<Path>,
+) -> Result<(), error::FilesystemError> {
+    let data = addons
+        .into_iter()
+        .map(|(flavor, addons)| {
+            (
+                flavor,
+                addons
+                    .into_iter()
+                    .map(|a| Data::try_from((flavor, a)))
```

Comment:
When we use `flavor` i feel like we should use `addon` rather than `a`.

```suggestion
                    .map(|addon| Data::try_from((flavor, addon)))
```